Index: src/chrome/browser/zenium_flag_choices.h
===================================================================
--- src.orig/chrome/browser/zenium_flag_choices.h
+++ src/chrome/browser/zenium_flag_choices.h
@@ -29,6 +29,9 @@ constexpr const FeatureEntry::Choice kCh
 
 constexpr const char kUpdateServerUrlCommandLine[] = "custom-update-server-url";
 
+constexpr const char kMiddleClickAutoscrollCommandLine[] =
+    "middle-click-autoscroll";
+
 }  // namespace zenium
 
 #endif /* CHROME_BROWSER_ZENIUM_FLAG_CHOICES_H_ */
Index: src/chrome/browser/zenium_flag_entries.h
===================================================================
--- src.orig/chrome/browser/zenium_flag_entries.h
+++ src/chrome/browser/zenium_flag_entries.h
@@ -23,6 +23,10 @@
      "Overrides the default URL for fetching updates. Should be used only for "
      "testing. Zenium flag.",
      kOsDesktop, STRING_VALUE_TYPE(zenium::kUpdateServerUrlCommandLine, "")},
+    {zenium::kMiddleClickAutoscrollCommandLine, "Middle Click Autoscroll",
+     "Enables autoscroll on middle click. Zenium flag, Chromium feature.",
+     kOsDesktop,
+     FEATURE_VALUE_TYPE(blink::features::kZeniumMiddleClickAutoscroll)},
     // core/network
     // ui
     // platform
Index: src/third_party/blink/public/common/features.h
===================================================================
--- src.orig/third_party/blink/public/common/features.h
+++ src/third_party/blink/public/common/features.h
@@ -40,6 +40,8 @@ BLINK_COMMON_EXPORT extern const base::F
 // `RuntimeEnabledFeatures)`, they should still be ordered in this section based
 // on the identifier name of the generated feature.
 
+BLINK_COMMON_EXPORT BASE_DECLARE_FEATURE(kZeniumMiddleClickAutoscroll);
+
 // Controls whether to include information about the page's open popup in
 // AIPageContent.
 BLINK_COMMON_EXPORT BASE_DECLARE_FEATURE(kAIPageContentIncludePopupWindows);
Index: src/third_party/blink/common/features.cc
===================================================================
--- src.orig/third_party/blink/common/features.cc
+++ src/third_party/blink/common/features.cc
@@ -19,6 +19,15 @@
 
 namespace blink::features {
 
+BASE_FEATURE(kZeniumMiddleClickAutoscroll,
+             "ZeniumMiddleClickAutoscroll",
+#if BUILDFLAG(IS_WIN)
+             base::FEATURE_ENABLED_BY_DEFAULT
+#else
+             base::FEATURE_DISABLED_BY_DEFAULT
+#endif
+);
+
 BASE_FEATURE(kDisableLinkDrag, "DisableLinkDrag", base::FEATURE_DISABLED_BY_DEFAULT);
 BASE_FEATURE(kReducedSystemInfo, "ReducedSystemInfo", base::FEATURE_DISABLED_BY_DEFAULT);
 BASE_FEATURE(kRemoveClientHints, "RemoveClientHints", base::FEATURE_DISABLED_BY_DEFAULT);
Index: src/third_party/blink/renderer/core/exported/web_view_impl.cc
===================================================================
--- src.orig/third_party/blink/renderer/core/exported/web_view_impl.cc
+++ src/third_party/blink/renderer/core/exported/web_view_impl.cc
@@ -1905,9 +1905,9 @@ void WebView::ApplyWebPreferences(const
       prefs.default_maximum_page_scale_factor);
 #endif
 
-#if BUILDFLAG(IS_WIN)
-  RuntimeEnabledFeatures::SetMiddleClickAutoscrollEnabled(true);
-#endif
+  RuntimeEnabledFeatures::SetMiddleClickAutoscrollEnabled(
+    base::FeatureList::IsEnabled(features::kZeniumMiddleClickAutoscroll)
+  );
 
   RuntimeEnabledFeatures::SetTranslateServiceEnabled(
       prefs.translate_service_available);
