Index: src/chrome/browser/renderer_context_menu/render_view_context_menu.cc
===================================================================
--- src.orig/chrome/browser/renderer_context_menu/render_view_context_menu.cc
+++ src/chrome/browser/renderer_context_menu/render_view_context_menu.cc
@@ -1291,17 +1291,6 @@ void RenderViewContextMenu::InitMenu() {
     AppendSpellingAndSearchSuggestionItems();
   }
 
-  if (content_type_->SupportsGroup(
-          ContextMenuContentType::ITEM_GROUP_AUTOFILL)) {
-    // Add a separator if the previous item is not already a separator.
-    size_t count = menu_model_.GetItemCount();
-    if (features::IsMenuSimplificationEnabled() && count > 0 &&
-        menu_model_.GetTypeAt(count - 1) != ui::MenuModel::TYPE_SEPARATOR) {
-      menu_model_.AddSeparator(ui::NORMAL_SEPARATOR);
-    }
-    autofill_context_menu_manager_.AppendItems();
-  }
-
   if (editable) {
     AppendOtherEditableItems();
   }
@@ -2484,49 +2473,6 @@ void RenderViewContextMenu::AppendPageIt
                                     IDS_CONTENT_CONTEXT_SAVEPAGEAS);
     menu_model_.AddItemWithStringId(IDC_PRINT, IDS_CONTENT_CONTEXT_PRINT);
 
-    AppendLiveCaptionItem();
-    menu_model_.AddSeparator(ui::NORMAL_SEPARATOR);
-
-    if (!glic_below_search) {
-      // Ask gemini
-      MaybeAppendOpenGlicItem(/*add_separator=*/false);
-
-      // Search with google lens
-      if (IsRegionSearchEnabled()) {
-        AppendRegionSearchItem();
-      }
-    } else {
-      // Search with google lens
-      if (IsRegionSearchEnabled()) {
-        AppendRegionSearchItem();
-      }
-
-      // Ask gemini
-      MaybeAppendOpenGlicItem(/*add_separator=*/false);
-    }
-
-    // Open in reading mode & Listen to this page
-    AppendReadAnythingItem();
-
-    // Cast
-    AppendMediaRouterItem();
-
-    // Send to your devices
-    if (GetBrowser() &&
-        send_tab_to_self::ShouldDisplayEntryPoint(embedder_web_contents_)) {
-      AppendSendTabToSelfItem(/*add_separator=*/false);
-    }
-
-    // Create QR code for this page
-    AppendQRCodeGeneratorItem(/*for_image=*/false, /*draw_icon=*/true,
-                              /*add_separator=*/false,
-                              /*ignore_simplification=*/true);
-
-    // Translate to language
-    if (CanTranslate(/*menu_logging=*/true)) {
-      AppendTranslateItem();
-    }
-
     menu_model_.AddSeparator(ui::NORMAL_SEPARATOR);
     return;
   }
@@ -2544,7 +2490,6 @@ void RenderViewContextMenu::AppendPageIt
                                   IDS_CONTENT_CONTEXT_SAVEPAGEAS);
   menu_model_.AddItemWithStringId(IDC_PRINT, IDS_CONTENT_CONTEXT_PRINT);
   AppendLiveCaptionItem();
-  AppendMediaRouterItem();
 
   if (features::IsReadAnythingMenuShuffleExperimentEnabled()) {
     // This will be set to false in AppendRegionSearchItem() if it is called.
@@ -2764,7 +2709,7 @@ void RenderViewContextMenu::AppendReadAn
   }
 
   // Show Read Anything option if it's not already open in the side panel.
-  if (IsNormalBrowser() && !IsReadAnythingEntryShowing(GetBrowser())) {
+  if ((false)) {
     std::u16string label;
     const std::u16string printable_selection_text = PrintableSelectionText();
     if (is_menu_simplification_enabled && !printable_selection_text.empty()) {
@@ -4178,30 +4123,9 @@ void RenderViewContextMenu::ExecuteComma
   }
 }
 
-void RenderViewContextMenu::AddSpellCheckServiceItem(bool is_checked) {
-  AddSpellCheckServiceItem(&menu_model_, is_checked);
-}
+void RenderViewContextMenu::AddSpellCheckServiceItem(bool) {}
 
-void RenderViewContextMenu::AddAccessibilityLabelsServiceItem(bool is_checked) {
-  if (is_checked) {
-    menu_model_.AddCheckItemWithStringId(
-        IDC_CONTENT_CONTEXT_ACCESSIBILITY_LABELS_TOGGLE,
-        IDS_CONTENT_CONTEXT_ACCESSIBILITY_LABELS_MENU_OPTION);
-  } else {
-    // Add the submenu if the whole feature is not enabled.
-    accessibility_labels_submenu_model_.AddItemWithStringId(
-        IDC_CONTENT_CONTEXT_ACCESSIBILITY_LABELS_TOGGLE,
-        IDS_CONTENT_CONTEXT_ACCESSIBILITY_LABELS_SEND);
-    accessibility_labels_submenu_model_.AddItemWithStringId(
-        IDC_CONTENT_CONTEXT_ACCESSIBILITY_LABELS_TOGGLE_ONCE,
-        IDS_CONTENT_CONTEXT_ACCESSIBILITY_LABELS_SEND_ONCE);
-    menu_model_.AddSubMenu(
-        IDC_CONTENT_CONTEXT_ACCESSIBILITY_LABELS,
-        l10n_util::GetStringUTF16(
-            IDS_CONTENT_CONTEXT_ACCESSIBILITY_LABELS_MENU_OPTION),
-        &accessibility_labels_submenu_model_);
-  }
-}
+void RenderViewContextMenu::AddAccessibilityLabelsServiceItem(bool) {}
 
 // static
 void RenderViewContextMenu::RegisterMenuShownCallbackForTesting(
@@ -4583,7 +4507,7 @@ bool RenderViewContextMenu::AppendQRCode
     return false;
   }
 
-  if (!IsQRCodeGeneratorEnabled()) {
+  if ((true)) {
     return false;
   }
 
@@ -5669,19 +5593,11 @@ ToastController* RenderViewContextMenu::
 }
 
 bool RenderViewContextMenu::CanTranslate(bool menu_logging) {
-  ChromeTranslateClient* chrome_translate_client =
-      ChromeTranslateClient::FromWebContents(embedder_web_contents_);
-  return chrome_translate_client &&
-         chrome_translate_client->GetTranslateManager()->CanManuallyTranslate(
-             menu_logging);
+  return false;
 }
 
 bool RenderViewContextMenu::CanPartiallyTranslateTargetLanguage() {
-  ChromeTranslateClient* chrome_translate_client =
-      ChromeTranslateClient::FromWebContents(embedder_web_contents_);
-  return chrome_translate_client &&
-         chrome_translate_client->GetTranslateManager()
-             ->CanPartiallyTranslateTargetLanguage();
+  return false;
 }
 
 void RenderViewContextMenu::MaybePrepareForLensQuery() {
Index: src/chrome/browser/ui/browser_tab_strip_model_delegate.cc
===================================================================
--- src.orig/chrome/browser/ui/browser_tab_strip_model_delegate.cc
+++ src/chrome/browser/ui/browser_tab_strip_model_delegate.cc
@@ -326,7 +326,7 @@ void BrowserTabStripModelDelegate::AddTo
 }
 
 bool BrowserTabStripModelDelegate::SupportsReadLater() {
-  return !browser_->profile()->IsGuestSession() && !IsForWebApp();
+  return false;
 }
 
 bool BrowserTabStripModelDelegate::IsForWebApp() {
