Index: src/chrome/browser/ui/ui_features.cc
===================================================================
--- src.orig/chrome/browser/ui/ui_features.cc
+++ src/chrome/browser/ui/ui_features.cc
@@ -245,13 +245,7 @@ bool IsTabGroupHoverCardsEnabled() {
 
 // Enables preview images in tab-hover cards.
 // https://crbug.com/41439486
-BASE_FEATURE(kTabHoverCardImages,
-#if BUILDFLAG(IS_MAC)
-             base::FEATURE_DISABLED_BY_DEFAULT
-#else
-             base::FEATURE_ENABLED_BY_DEFAULT
-#endif
-);
+BASE_FEATURE(kTabHoverCardImages, base::FEATURE_ENABLED_BY_DEFAULT);
 
 // Skips the synthetic tab selection event fired when a browser window is
 // activated.
