Index: src/chrome/browser/ui/omnibox/omnibox_view.cc
===================================================================
--- src.orig/chrome/browser/ui/omnibox/omnibox_view.cc
+++ src/chrome/browser/ui/omnibox/omnibox_view.cc
@@ -157,10 +157,6 @@ ui::ImageModel OmniboxView::GetIcon(int
         return ui::ImageModel::FromImage(icon);
       }
     }
-    favicon = turl ? controller()->client()->GetFaviconForKeywordSearchProvider(
-                         turl, std::move(on_icon_fetched))
-                   : controller()->client()->GetFaviconForDefaultSearchProvider(
-                         std::move(on_icon_fetched));
 
   } else if (match.type != AutocompleteMatchType::HISTORY_CLUSTER) {
     // The starter pack suggestions are a unique case. These suggestions
