Index: src/chrome/browser/chrome_content_browser_client.cc
===================================================================
--- src.orig/chrome/browser/chrome_content_browser_client.cc
+++ src/chrome/browser/chrome_content_browser_client.cc
@@ -350,6 +350,7 @@
 #include "content/public/browser/internal_webui_config.h"
 #include "content/public/browser/isolated_web_apps_policy.h"
 #include "content/public/browser/legacy_tech_cookie_issue_details.h"
+#include "content/public/browser/navigation_entry.h"
 #include "content/public/browser/navigation_handle.h"
 #include "content/public/browser/network_service_instance.h"
 #include "content/public/browser/overlay_window.h"
Index: src/chrome/browser/extensions/BUILD.gn
===================================================================
--- src.orig/chrome/browser/extensions/BUILD.gn
+++ src/chrome/browser/extensions/BUILD.gn
@@ -1049,6 +1049,7 @@ source_set("extensions") {
       "//components/resources",
       "//components/safe_browsing:buildflags",
       "//components/safe_browsing/content/browser/web_ui",
+      "//components/safe_browsing/content/common/proto:download_file_types_proto",
       "//components/safe_browsing/core/common/proto:csd_proto",
       "//components/search_engines",
       "//components/services/unzip/public/cpp",
Index: src/chrome/common/BUILD.gn
===================================================================
--- src.orig/chrome/common/BUILD.gn
+++ src/chrome/common/BUILD.gn
@@ -532,6 +532,7 @@ static_library("url_constants") {
     "//components/optimization_guide/optimization_guide_internals/webui:url_constants",
     "//components/password_manager/content/common",
     "//components/safe_browsing/core/common",
+    "//components/supervised_user/core/common:buildflags",
     "//device/vr/buildflags",
   ]
 
Index: src/components/component_updater/installer_policies/BUILD.gn
===================================================================
--- src.orig/components/component_updater/installer_policies/BUILD.gn
+++ src/components/component_updater/installer_policies/BUILD.gn
@@ -42,6 +42,7 @@ static_library("installer_policies_no_co
     "//components/update_client",
     "//mojo/public/cpp/base:protobuf_support",
     "//services/network/public/cpp",
+    "//third_party/re2",
   ]
 
   # Disallow depending on content.
Index: src/components/contextual_tasks/internal/BUILD.gn
===================================================================
--- src.orig/components/contextual_tasks/internal/BUILD.gn
+++ src/components/contextual_tasks/internal/BUILD.gn
@@ -51,6 +51,7 @@ source_set("internal") {
     "//components/sync/model",
     "//components/url_formatter",
     "//net",
+    "//third_party/re2",
     "//ui/gfx",
     "//url",
   ]
Index: src/content/browser/BUILD.gn
===================================================================
--- src.orig/content/browser/BUILD.gn
+++ src/content/browser/BUILD.gn
@@ -144,6 +144,7 @@ source_set("browser") {
     "//components/filename_generation",
     "//components/history/core/browser:features",
     "//components/input",
+    "//components/lens:buildflags",
     "//components/language/core/common",
     "//components/language_detection/content/common",
     "//components/link_header_util",
